### DISCUSSION TITLE
Can't type in inputs on release build

### DISCUSSION THREAD

**Discussion by NinjaJake86** [tauri-apps/tauri#3541] [Q&A]
Hi All,

Developing a Tauri App with Vue 3 and Element Plus, all works fine while debugging but once i install a release i can no longer type in text fields, has anyone else seen this before and know of a solution?

Thanks!

**Comment by FabianLars** [comment]
Can't remember anyone with that specific problem 🤔 
Can you check the common pitfalls?
- Are all your dependencies up to date?
- Does the same happen with a debug build? `tauri build --debug` -> output in target/debug/
- output of `tauri info`?
- enable devtools in your release build and check the console for errors: add the `"devtools"` feature flag to the tauri dep in Cargo.toml
- check if it's not a frontend dev vs prod thingy is (depends on your frontend&bunndler)

**Reply by NinjaJake86** [reply]
Thanks for the help!

1. Updated, no change
2. Same happens with debug build
3. Tauri info output below

```
$ tauri info

Operating System - Windows, version 10.0.22000 X64
Webview2 - 98.0.1108.56
Visual Studio Build Tools:
   - Visual Studio Build Tools 2019
   - Visual Studio Professional 2017

Node.js environment
  Node.js - 16.13.0
  @tauri-apps/cli - 1.0.0-rc.5
  @tauri-apps/api - 1.0.0-rc.1

Global packages
  npm - 8.3.0
  pnpm - 6.11.0
  yarn - 1.22.15

Rust environment
  rustup - 1.24.3
  rustc - 1.58.1
  cargo - 1.58.0
  toolchain - stable-x86_64-pc-windows-msvc 

App directory structure
/.vscode
/dist
/node_modules
/public
/src
/src-tauri

App
  tauri - 1.0.0-rc.2
  tauri-build - 1.0.0-rc.2
  tao - 0.6.2
  wry - 0.13.1
  build-type - bundle
  CSP - default-src http://sphere.smpe.com/ blob: data: filesystem: ws: wss: http: https: tauri: 'unsafe-eval' 'unsafe-inline' 'self' img-src: 'self'
  distDir - ../dist
  devPath - http://localhost:3000/
  framework - Vue.js
Done in 21.44s.
```

4. Ran devtools and found errors indicating that the v-model values didn't exist or couldn't be found which is weird, i've included the code for this page below, any ideas?

```

<script setup>
  import { CircleCheck } from "@element-plus/icons-vue";
  import { getClient, Body } from "@tauri-apps/api/http";
  import { ElMessage } from "element-plus";
</script>

<template>
  <div style="width: 100vw; height: 100vh; margin: 0; padding: 0">
    <div
      style="
        width: 15rem;
        height: 20rem;
        padding: 0;
        margin: auto;
        margin-top: calc(50vh - 10rem);
        display: flex;
        flex-direction: column;
      "
    >
      <el-icon size="50" style="margin-left: auto; margin-right: auto"
        ><circle-check
      /></el-icon>
      <h1 style="text-align: center; font-size: x-large">Sphere</h1>
      <el-input v-model="email" placeholder="Email" style="margin-top: 0.5rem" />
      <el-input
        v-model="password"
        placeholder="Password"
        type="password"
        style="margin-top: 0.5rem"
      />
      <el-button type="primary" style="margin-top: 1rem; width: 100%" @click="login"
        >Login</el-button
      >
    </div>
  </div>
</template>

<script>
  export default {
    data() {
      return{
        email: "",
        password: ""
      }
    },
    methods: {
      login: async function() {
          const client = await getClient();

          client
            .request({
              url: "http://***/api/login",
              method: "POST",
              headers: {
                Accept: "application/json, text/plain, */*",
                "Content-Type": "application/json",
              },
              body: Body.json({ email: this.email, password: this.password}),
            })
            .then((response) => {
              if (response.ok) {
                let data = response.data;
                if(data.status == 'success') {
                  this.$store.commit('apikey', {key: data.key});
                  this.$store.commit("loginUser", { name: data.name, department: data.department, widgets: data.widgets, apps: data.apps, sideApps: data.sideApps});
                  this.$router.push({ name: 'Dashboard' });
                } else ElMessage.error('Login Failed');
              }
              else ElMessage.error('Login Failed');
            })
            .catch((error) => {
              ElMessage.error(error.toString());
            });
        },
    }
  };
</script>

```

**Reply by dchicchon** [reply]
@FabianLars when you specify the `devtools` feature flag, where did you find the documentation for this? Can't seem to find it anywhere

**Comment by dceddia** [comment]
I think I hit the same problem, or at least a similar one. I couldn't type into input fields in dev or release mode. So, while the app was running in dev mode, I tried opening it directly in Firefox, Chrome, and Safari (`http://localhost:3000`). With Firefox and Chrome I could type into the inputs just fine. With Safari, it was the same behavior as Tauri: couldn't type, and when I focused the input, it would take focus but wouldn't show the text cursor.

So I went googling and found [this post](https://armno.in.th/2020/11/26/css-user-select/) about fixing a similar bug on iOS Safari by adding the CSS style `-webkit-user-select: text` to the input – and lo and behold, it worked!

I doubt this is a Tauri thing because I was hitting my Vite devserver directly, so maybe it's something else in my stack. I'm using Svelte, Tailwind, TypeScript, building with Vite. Dunno. `-webkit-user-select: text` for the win, though.

EDIT: Figured it out. I had this block of overzealous code in my index.css:

```css
/* Make it more native-ish */
*,
*::after,
*::before {
  user-select: none;
  -webkit-user-select: none;
  -webkit-user-drag: none;
  -webkit-app-region: no-drag;
  cursor: default;
}
```

I added this below it to bring selection & cursor behavior back to normal for inputs:

```css
input {
  user-select: initial;
  -webkit-user-select: initial;
  cursor: initial;
}
```

**Reply by alfreddagenais** [reply]
Thanks you @dceddia this work for me. 

tailwindcss (3.1.8) + svelte (3.49.0) + @tauri-apps (1.0.5)

**Reply by dceddia** [reply]
Nice! Also a good idea to add textarea to that selector too… that one bit me a couple months after this 😂

**Reply by ahmadyusufcodes** [reply]
Thanks bruv! Added it to end of my global.css and 🚀

**Reply by ProfNuru** [reply]
This solution did not work for me. After debugging, I found that changing the input type to password stops the flickering. I still don't know how to fix this issue though. 
I am on Windows 11 and running tauri with react and vite.
Edit: Setting `autoComplete="off"` fixed it
Edit 2: Nope! It did not fix it. Only fix it until I re-rendered. SMH!!

**Reply by dceddia** [reply]
Flickering sounds like a different issue. This was more like it wouldn’t accept keypresses at all.

The change after a rerender seems like a clue though… maybe the input’s props are changing unexpectedly, like flipping between rendering an empty value and another value before settling.

edit: this reminds me a little of times when the input has been "off by one" or seems to be out of sync with the keypresses, and if that's what's going on it can be maddening to figure out the order of things because there's only 2 states and it's either early or late 😅 I'd make sure the state change/whatever is only happening once, and that it's happening immediately on keypress, not delayed by a useEffect or timeout or some other thing.

edit 2: if what you're seeing is something like: you press "a", then it shows "a" for a split second and then goes back to blank, that'll be something like you're not updating the input's `value` prop or somehow setting it to blank after a change. Check the `onChange` handler if you have one.

**Reply by ProfNuru** [reply]
Nope. Here's a video of what I'm talking about:

https://github.com/user-attachments/assets/0fe8942e-8827-430f-bd7e-f7a48ec9f403

It happens even when I do not handle it with state values. When I open the frontend in a browser, it works perfectly. So it has to be from the webview in tauri.
Notice that sometimes it doesn't happen.

**Reply by dceddia** [reply]
Ok strange! I agree it looks like it's trying to autocomplete. Maybe some sort of browser extension? 🤔 But then it seems like it would affect Edge too.

That's definitely different than the behavior I saw when I left the original comment here - mine acted like if you set up the `onChange` handler without setting a `value`, where keypresses were just completely ignored. I'd probably file a separate issue or check the Discord.

**Reply by ProfNuru** [reply]
I have filed an issue already: https://github.com/tauri-apps/tauri/issues/12928#issuecomment-2707996199

**Reply by azlouk** [reply]
hello bro , i get a  same problem and finally i resolve it by change web view version old to  WebView2: 133.0.3065134.0.3124.51 (Microsoft.WebView2.FixedVersionRuntime.133.0.3065.92.x64.cab) my current is  WebView2: 134.0.3124.51 then i set it in variable enveromment  by power shell run it as administrator and tape this :
[System.Environment]::SetEnvironmentVariable("WEBVIEW2_BROWSER_EXECUTABLE_FOLDER", "C:\WebView2Fixed", "Machine")
Start-Process "C:\WebView2Fixed\msedgewebview2.exe"
Attention should extract Microsoft.WebView2.FixedVersionRuntime.133.0.3065.92.x64.cab and change path C:\WebView2Fixed by your path
Enjoy

**Reply by FlazeIGuess** [reply]
@azlouk How exactly should i use the cmd? Should i download the other version of WebView2 first?

**Reply by azlouk** [reply]
Of course should download file then extract using winrar then execute list of commands with power shell

**Reply by azlouk** [reply]
https://developer.microsoft.com/fr-fr/microsoft-edge/webview2?form=MA13LH#download![Screenshot_20250316_211013_Chrome.jpg](https://github.com/user-attachments/assets/4ff5b583-8133-43b7-ba85-e74fabbab440)

**Comment by FlazeIGuess** [comment]
I found another solution thanks to @azlouk comment.

Download WebView2 fixed version: https://developer.microsoft.com/en-us/microsoft-edge/webview2?form=MA13LH'

this should be a .cab file

put the .cab file inside your Tauri Project, root folder

use this command:
Expand .\"CHANGE TO THE .cab FILE NAME" -F:* ./src-tauri

after that add to your tauri.conf.json:
"bundle": {
  "windows": {
        "webviewInstallMode": {
          "type": "fixedRuntime",
          "path": "./Microsoft.WebView2.FixedVersionRuntime.133.0.3065.92.x64/" <--- Change to the name of the folder inside src-tauri
        }
      },
      
Voila, should work now :)
